### PR TITLE
intersect-resources: Force immediate first measure for ads

### PR DESCRIPTION
**Pull request by choumx**
Partial for #25428. 

Reduce time-to-first-measure for ads (only) by manually measuring instead of waiting for the client rect from the `IntersectionObserver` callback.

Also allow measure requests from owned elements (probably an oversight).

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -1298,17 +1292,20 @@ export class ResourcesImpl {
...
           dev().fine(TAG_, 'force remeasure:', r.debugid);
         }
-        const premeasured = r.hasBeenPremeasured();
-        const needsMeasure = premeasured || requested || this.relayoutAll_;
+        // Immediately measure (vs. waiting for async premeasure) for certain
+        // elements with sensitive time-to-measure.
+        const expediteFirstMeasure =
+          !r.hasBeenMeasured() && r.element.tagName == 'AMP-AD';
+        const needsMeasure =
+          premeasured || requested || this.relayoutAll_ || expediteFirstMeasure;
```
Comment (choumx) [inline src/service/resources-impl.js:1308]: FYI these changes bring the experiment path closer to the non-experiment path: https://github.com/ampproject/amphtml/pull/29581/files#diff-80b9cea934a00a64f15d14cb656dfad4R1339-R1344

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -461,11 +461,7 @@ export class ResourcesImpl {
       resource.getState() != ResourceState.NOT_BUILT &&
       !element.reconstructWhenReparented()
     ) {
-      // With IntersectionObserver, no need to request remeasure
-      // on reuse since initial intersection callback will trigger soon.
-      if (!this.intersectionObserver_) {
-        resource.requestMeasure();
-      }
+      resource.requestMeasure();
```
Comment (choumx) [inline src/service/resources-impl.js:464]: Not sure this actually matters in practice but reducing divergence just in case.

**Comment by samouri** [comment]
I remember you explained it in standup, but can you write out the hypothetical edge case this addresses in the description for the PR?

**Comment by choumx** [comment]
Yup!

**Pull request merged**